plugins/modules/cohesity_uda_job.py

```diff
@@ -0,0 +1,702 @@
+    description:
+      - "Existing Storage Domain to which the Protection Job will be associated. Required when I(state=present)."
+    type: str
+  time_zone:
+    default: America/Los_Angeles
+    description:
+      - "Specifies the timezone to use when calculating time for this Protection Job such as the Job start time."
+    type: str
+  validate_certs:
+    default: false
+    description:
+      - "Switch determines if SSL Validation should be enabled."
+    type: bool
+extends_documentation_fragment:
+- cohesity.dataprotect.cohesity
+short_description: "Management of Cohesity Protection Jobs"
+version_added: 1.0.3
+"""
+
+EXAMPLES = """
+# Create a new UDA Protection Job
+- cohesity_uda_job:
+    cluster: cohesity.lab
+    username: admin
+    password: password
+    state: present
+    name: myhost
+    endpoint: cohesity-source-ip
+    protection_policy: Bronze
+    storage_domain: Default
+
+"""
+
+RETURN = """
+# Returns the registered Protection Job ID
+"""
+
+import json
+import time
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url, urllib_error
+
+
+from cohesity_management_sdk.exceptions.api_exception import APIException
+from cohesity_management_sdk.models.delete_protection_job_param import (
+    DeleteProtectionJobParam,
+)
+from cohesity_management_sdk.models.cancel_protection_job_run_param import (
+    CancelProtectionJobRunParam,
+)
+from cohesity_management_sdk.models.run_protection_job_param import (
+    RunProtectionJobParam,
+)
+
+
+try:
+    # => When unit testing, we need to look in the correct location however, when run via ansible,
+    # => the expectation is that the modules will live under ansible.
+    from ansible.module_utils.urls import urllib_error
+    from ansible_collections.cohesity.dataprotect.plugins.module_utils.cohesity_auth import (
+        get__cohesity_auth__token,
+    )
+    from ansible_collections.cohesity.dataprotect.plugins.module_utils.cohesity_utilities import (
+        cohesity_common_argument_spec,
+        raise__cohesity_exception__handler,
+        REQUEST_TIMEOUT,
+    )
+    from ansible_collections.cohesity.dataprotect.plugins.module_utils.cohesity_hints import (
+        get_cohesity_client,
+        get__prot_source_id__by_endpoint,
+    )
+except Exception:
+    pass
+
+
+class ParameterViolation(Exception):
+    pass
+
+
+class ProtectionException(Exception):
+    pass
+
+
+def get_timezone():
+    # Function to get ansible control node timezone.
+    # :returns timezone
+    default_timezone = "America/Los_Angeles"
+    try:
+        import subprocess
+
+        cmd = "timedatectl status"
+        proc = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, stdin=subprocess.PIPE, shell=True
+        )
+        out, err = proc.communicate()
+        for line in out.split("\n"):
+            if "Time zone" in line:
+                default_timezone = line.split()[2]
+    except Exception:
+        pass
```

Comment:
Do not leave exception handling this way, use ansible modules to catch exceptions or to throw valid ones.